lighthouse-core/test/audits/audit-test.js

```diff
@@ -37,64 +61,94 @@ describe('Audit', () => {
+    it('throws if an audit returns a score > 1', () => {
+      assert.throws(_ => Audit.generateAuditResult(PassOrFailAudit, {score: 100}), /is > 1/);
+      assert.throws(_ => Audit.generateAuditResult(PassOrFailAudit, {score: 2}), /is > 1/);
+    });
+
+    it('throws if an audit returns a score < 0', () => {
+      assert.throws(_ => Audit.generateAuditResult(PassOrFailAudit, {score: -0.1}), /is < 0/);
+      assert.throws(_ => Audit.generateAuditResult(PassOrFailAudit, {score: -100}), /is < 0/);
     });
 
     it('throws if an audit returns a score that\'s not a number', () => {
       const re = /Invalid score/;
-      assert.throws(_ => Audit._normalizeAuditScore(B, {rawValue: true, score: NaN}), re);
-      assert.throws(_ => Audit._normalizeAuditScore(B, {rawValue: true, score: 'string'}), re);
-      assert.throws(_ => Audit._normalizeAuditScore(B, {rawValue: true, score: 50}), /is > 1/);
+      assert.throws(_ => Audit.generateAuditResult(PassOrFailAudit, {score: NaN}), re);
+      assert.throws(_ => Audit.generateAuditResult(PassOrFailAudit, {score: 'string'}), re);
     });
-  });
 
-  describe('generateAuditResult', () => {
-    it('throws if an audit does return a result with a rawValue', () => {
-      assert.throws(_ => Audit.generateAuditResult(B, {}), /requires a rawValue/);
+    it('throws if an audit does not return a result with a score', () => {
+      assert.throws(_ => Audit.generateAuditResult(PassOrFailAudit, {}), /requires a score/);
+    });
+
+    it('clamps the score to two decimals', () => {
+      const auditResult = Audit.generateAuditResult(PassOrFailAudit, {score: 0.29666666666666663});
+      assert.strictEqual(auditResult.score, 0.3);
+    });
+
+    it('chooses the title if score is passing', () => {
+      const auditResult = Audit.generateAuditResult(PassOrFailAudit, {score: 1});
+      assert.strictEqual(auditResult.score, 1);
+      assert.equal(auditResult.title, 'Passing');
     });
 
     it('chooses the failureTitle if score is failing', () => {
-      class FailingAudit extends Audit {
-        static get meta() {
-          return {
-            title: 'Passing',
-            failureTitle: 'Failing',
-          };
-        }
-      }
-
-      const auditResult = Audit.generateAuditResult(FailingAudit, {rawValue: false});
-      assert.ok(Number.isFinite(auditResult.score));
-      assert.equal(auditResult.score, 0);
+      const auditResult = Audit.generateAuditResult(PassOrFailAudit, {score: 0});
+      assert.strictEqual(auditResult.score, 0);
       assert.equal(auditResult.title, 'Failing');
     });
+
+    it('chooses the title if audit is not scored due to scoreDisplayMode', () => {
+      const auditResult = Audit.generateAuditResult(PassOrFailAudit,
+          {score: 0, errorMessage: 'what errors lurk'});
+      assert.strictEqual(auditResult.score, null);
+      assert.equal(auditResult.title, 'Passing');
```

Comment:
we use the passing title for errors? funny the things you learn in testing :)

Reply:
> we use the passing title for errors? funny the things you learn in testing :)

yeah, I was really surprised when I made `score === null` explicitly use the `failureTitle` and a whole bunch of things changed in `sample_v2.json` :)

I'm not entirely convinced it was on purpose, but it makes sense for `NOT_APPLICABLE` and `INFORMATIVE`, and it kind of makes sense for `ERROR`, since we don't know the page was doing the failure thing.